sources/main.functions.php

```diff
@@ -2590,15 +2587,19 @@ function storeUsersShareKey(
             $post_object_id
         );
     }
+
+    // Get the user ID
+    $userId = ($apiUserId === -1) ? (int) $session->get('user-id') : $apiUserId;
     
+    // $onlyForUser is only dynamically set by external calls
     if (
         $onlyForUser === true || (int) $post_folder_is_personal === 1
     ) {
         // Only create the sharekey for a user
         $user = DB::queryFirstRow(
             'SELECT public_key
             FROM ' . prefixTable('users') . '
-            WHERE id = ' . ($apiUserId === -1 ? (int) $session->get('user-id') : $apiUserId) . '
+            WHERE id = ' . $userId . '
```

Comment:
Can you replace this concat by %i or explicit int cast?

Reply:
of course